### PR TITLE
world.fps and SStimers safety check

### PR DESCRIPTION
**Pull request by Tsar-Salat**
Ports:
- https://github.com/tgstation/TerraGov-Marine-Corps/pull/2295
- https://github.com/tgstation/tgstation/pull/45236

## About The Pull Request

 if you actually change world.fps without a check like this you can have SStimer lists runtiming out of bounds, breaking a lot of things.

## Testing Photographs and Procedure

<details>
<summary>Screenshots&Videos</summary>

image

</details>

## Changelog
:cl: RKz, Rohesie
code: SStimers wont runtime out of bounds if you change world.fps
/:cl:

### CODE REVIEW

**Review by itsmeow — APPROVED** [review]

**Comment by Tsar-Salat** [comment]
This was technically an uncredited addition to #7285, which was later reverted for related issues.

Needless to say, I think those changes are good, but we should individually check which ported pr of that actually broke the server

File: code/game/world.dm [hunk 1]
```diff
@@ -415,6 +415,29 @@ GLOBAL_VAR(restart_counter)
...
+	if(new_value <= 0)
+		CRASH("change_fps() called with [new_value] new_value.")
+	if(fps == new_value)
+		return //No change required.
+
+	fps = new_value
+	on_tickrate_change()
+
+
+/world/proc/change_tick_lag(new_value = 0.5)
```
Comment (itsmeow) [inline code/game/world.dm:424]: is this even used anywhere?

Comment (Tsar-Salat) [reply code/game/world.dm:424]: Nope, nothing calls it. Not even tg uses it.

Comment (Tsar-Salat) [reply code/game/world.dm:424]: ill just comment it for now. aec22d0

Comment (itsmeow) [reply code/game/world.dm:424]: I mean is tick_lag set anywhere else

Comment (Tsar-Salat) [reply code/game/world.dm:424]: No, not from what I can see. Looks just to be some "incase" code

**Comment by Tsar-Salat** [comment]
im taking a break, Semester begins soon and this is affecting my mental health

**Pull request merged**